### ISSUE TITLE
mutex issues with TtlSet

### ISSUE THREAD

**Issue by broamski** [ChannelMeter/vault-gatekeeper-mesos#38]
It appears that locking is not operating as expected on the `TtlSet` struct. I was able to replicate this by having multiple forked processed in my application request a token at the same time:
```
2017/04/26 20:07:10 Provided token pair for x.x.x.x:xxxx in 343.859137ms. (Task Id: task-id-1234) (Task Name: application-name). Policies: [application_policy]
2017/04/26 20:07:10 Provided token pair for x.x.x.x:xxxx in 419.929055ms. (Task Id: task-id-1234) (Task Name: application-name). Policies: [application_policy]                                                                         
2017/04/26 20:07:10 Provided token pair for x.x.x.x:xxxx in 484.607434ms. (Task Id: task-id-1234) (Task Name: application-name). Policies: [application_policy]
2017/04/26 20:07:10 Provided token pair for x.x.x.x:xxxx in 656.629353ms. (Task Id: task-id-1234) (Task Name: application-name). Policies: [application_policy]
```

**Comment by nemosupremo** [comment]
Interesting. In hindsight, this bug obvious. There needs to be a per-task-id lock, meaning a request with the same task id shouldn't be served in parallel.

**Comment by broamski** [comment]
This was resolved with #44.
